### PR TITLE
Delete user account 

### LINKED ISSUE #421: Delete user account
- Waiting on API

### LINKED ISSUE #729: Implement delete user

### PR DESCRIPTION
**Pull request by sunyatax**
- Closes #421 
- Closes #729 

Hey @orestesgaolin - updated the UserProfile model in user_model.dart to accommodate emails. Noticed we have an automatically generated user_model.g.dart file as well. Is this also through freezed / and will I just need to call `flutter pub pub run build_runner build` ?

### CODE REVIEW

**Comment by orestesgaolin** [comment]
exactly, run `flutter pub pub run build_runner build` or sometimes if it doesn't work `pub run build_runner build --delete-conflicting-outputs` (which is a bit longer)

File: lib/backend/repositories/auth_repo.dart [hunk 1]
```diff
@@ -129,4 +129,9 @@ class AuthRepo {
     themesProvider.setTheme("rainbow");
     await authService.logoutUser();
   }
+
+  // Delete user account
+  Future<void> deleteUserAccount(String userAddress, String password) async {
```
Comment (Nash0x7E2) [inline lib/backend/repositories/auth_repo.dart:134]: Think we should add some assertion checks here to ensure the userAddress and password are not null or empty.

File: lib/backend/services/auth_service.dart [hunk 2]
```diff
@@ -158,6 +158,19 @@ class AuthenticationServiceCentralized implements AuthenticationService {
...
     final Map<String, dynamic> _responseMap =
         JuntoHttp.handleResponse(response);
-    print(_responseMap);
+  }
+
+  Future<void> deleteUserAccount(String userAddress, String password) async {
+    final Map<String, dynamic> _body = <String, String>{
+      'password': password,
+    };
+    print('first step');
```
Comment (Nash0x7E2) [inline lib/backend/services/auth_service.dart:167]: We should update to use `logger.logMessage` instead

File: lib/widgets/dialogs/delete_account_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        ),
+        decoration: BoxDecoration(
+          color: Theme.of(context).backgroundColor,
+          borderRadius: BorderRadius.circular(15),
+        ),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: <Widget>[
+            Text(
+              "If you are sure you'd like to delete your account, type your password below and confirm.",
```
Comment (Nash0x7E2) [inline lib/widgets/dialogs/delete_account_dialog.dart:41]: These string should be added to `Localizy`

File: lib/widgets/end_drawer/junto_account.dart [hunk 4]
```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:junto_beta_mobile/backend/backend.dart';
+import 'package:junto_beta_mobile/widgets/dialogs/delete_account_dialog.dart';
+
+import 'junto_account_appbar.dart';
```
Comment (Nash0x7E2) [inline lib/widgets/end_drawer/junto_account.dart:7]: Avoid file mixing packages and file imports. Dart will treat these as separate types which can cause problems

Comment (orestesgaolin) [reply lib/widgets/end_drawer/junto_account.dart:7]: This rule doesn't apply here. Within the same lib both imports work https://dart.dev/guides/language/effective-dart/usage#do-use-relative-paths-when-importing-libraries-within-your-own-packages-lib-directory

**Pull request merged**